<!-- -->
<launch>
	<arg name="maxVelocity" 				default="1.0" />
	<arg name="minVelocity" 				default="0.0" />
	<arg name="velocityProfileFactor" 		default="1.0" />
	<arg name="maxPlanningDistance" 		default="1000000.0" />
	<arg name="maxLocalPlanDistance" 		default="10.0" />
	<arg name="samplingTipMargin" 			default="1.5" />
	<arg name="samplingOutMargin" 			default="4.0" />
	<arg name="samplingSpeedFactor" 		default="0.25" />
	<arg name="pathDensity" 				default="0.25" />
	<arg name="rollOutDensity" 				default="0.25" />
	<arg name="rollOutsNumber" 				default="6" />
	<arg name="horizonDistance" 			default="15" />
	
	<arg name="minFollowingDistance" 		default="5.0" />
	<arg name="maxFollowingDistance" 		default="7.0" />
	<arg name="minDistanceToAvoid" 			default="3.0" />
	<arg name="speedProfileFactor"			default="1.0" />
	
	<arg name="enableSwerving" 				default="1" />
	<arg name="enableFollowing" 			default="1" />
	<arg name="enableHeadingSmoothing" 		default="0" />
	<arg name="enableTrafficLightBehavior" 	default="0" />
	<arg name="enableLaneChange" 			default="0" />
	
	<arg name="width" 						default="0.8" />
	<arg name="length" 						default="0.9" />
	<arg name="wheelBaseLength" 			default="0.6" />
	<arg name="turningRadius"				default="1.5" />
	<arg name="maxSteerAngle" 				default="0.75" />
	
	<arg name="planner" 					default="DP" />
	<arg name="system" 						default="segway" />
	<arg name="map" 						default="kml" />
	<arg name="mapDirectory" 				default="/home/hatem/SimuLogs/KmlMaps/7th floor 100.kml" />
	
	
	<node pkg="dp_planner" type="dp_planner" name="dp_planner" output="screen">
		<param name="maxVelocity" 					value="$(arg maxVelocity)" />
	    <param name="minVelocity" 					value="$(arg minVelocity)" />
	    
	    <param name="velocityProfileFactor" 		value="$(arg velocityProfileFactor)" />
		<param name="maxPlanningDistance" 			value="$(arg maxPlanningDistance)" />
		<param name="maxLocalPlanDistance" 			value="$(arg maxLocalPlanDistance)" />
		<param name="samplingTipMargin" 			value="$(arg samplingTipMargin)" />
		<param name="samplingOutMargin" 			value="$(arg samplingOutMargin)" />
		<param name="samplingSpeedFactor" 			value="$(arg samplingSpeedFactor)" />
		<param name="pathDensity" 					value="$(arg pathDensity)" />
		<param name="rollOutDensity" 				value="$(arg rollOutDensity)" />
		<param name="rollOutsNumber" 				value="$(arg rollOutsNumber)" />
		<param name="horizonDistance" 				value="$(arg horizonDistance)" />
		
		<param name="minFollowingDistance" 			value="$(arg minFollowingDistance)" />
		<param name="maxFollowingDistance" 			value="$(arg maxFollowingDistance)" />
		<param name="minDistanceToAvoid" 			value="$(arg minDistanceToAvoid)" />
		<param name="speedProfileFactor"			value="$(arg speedProfileFactor)" />
		
		<param name="enableSwerving" 				value="$(arg enableSwerving)" />
		<param name="enableFollowing" 				value="$(arg enableFollowing)" />
		<param name="enableHeadingSmoothing" 		value="$(arg enableHeadingSmoothing)" />
		<param name="enableTrafficLightBehavior" 	value="$(arg enableTrafficLightBehavior)" />
		<param name="enableLaneChange" 				value="$(arg enableLaneChange)" />
		
		<param name="width" 						value="$(arg width)" />
		<param name="length" 						value="$(arg length)" />
		<param name="wheelBaseLength" 				value="$(arg wheelBaseLength)" />
		<param name="turningRadius" 				value="$(arg turningRadius)" />
		<param name="maxSteerAngle" 				value="$(arg maxSteerAngle)" />
		
		<param name="planner" 						value="$(arg planner)" />
		<param name="system" 						value="$(arg system)" />
		<param name="map" 							value="$(arg map)" />
		<param name="mapDirectory" 					value="$(arg mapDirectory)" />
	    
	    
	</node>

</launch>